packages/trpc/server/routers/viewer/organizations/updateUser.handler.ts

```diff
@@ -72,6 +90,21 @@ export const updateUserHandler = async ({ ctx, input }: UpdateUserOptions) => {
     }),
   ]);
 
+  if (input.role === MembershipRole.ADMIN || input.role === MembershipRole.OWNER) {
+    await prisma.membership.updateMany({
+      where: {
+        userId: input.userId,
+        teamId: {
+          in: requestedMember.team.children.map((sub_team) => {
+            return sub_team.members.find((item) => item.userId === input.userId).teamId;
+          }),
+        },
+      },
+      data: {
+        role: input.role,
+      },
+    });
+  }
```

Comment:
How about abstracting this in a separate function `applyRoleToAllTeams` ? It would make the code easy to understand and testable if needed.

Reply:
Absolutely

Reply:
Approved it for now and it's ready to merge. You can push it in a separate PR, would be a quick merge.